common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -60,9 +66,52 @@ private void saveAccessToken(
             final AuthorizationRequest request,
             final TokenResponse response) {
         final AccessToken accessToken = mAccountCredentialAdapter.createAccessToken(oAuth2Strategy, request, response);
+        deleteAccessTokensWithIntersectingScopes(accessToken);
```

Comment:
I'm probably lack of context here. Why do we need to delete the access token of intersecting scope before saving the access token?

Reply:
The v2 endpoint always returns all of the user consented scopes associated with a particular resource.  As a result if the scopes intersect then the existing access token can be replaced.... or simply deleted.